server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundSummaryServiceTests.java

```diff
@@ -253,4 +278,173 @@ public void testEnableAndThenDisableService() {
+            Map.of(DUMMY_NODE, new DesiredBalanceMetrics.NodeWeightStats(10, 20, 30, 40)),
+            DesiredBalance.ComputationFinishReason.CONVERGED
+        );
+        // Add a new node and move one shard.
+        DesiredBalance secondDesiredBalance = new DesiredBalance(
+            1,
+            Map.of(new ShardId(INDEX_NAME, INDEX_UUID, 0), new ShardAssignment(Set.of(SECOND_DUMMY_NODE.getId()), 1, 0, 0)),
+            Map.of(
+                DUMMY_NODE,
+                new DesiredBalanceMetrics.NodeWeightStats(20, 40, 60, 80),
+                SECOND_DUMMY_NODE,
+                new DesiredBalanceMetrics.NodeWeightStats(5, 15, 25, 35)
+            ),
+            DesiredBalance.ComputationFinishReason.CONVERGED
+        );
+
+        var summary = AllocationBalancingRoundSummaryService.createBalancerRoundSummary(firstDesiredBalance, secondDesiredBalance);
+
+        assertEquals(1, summary.numberOfShardsToMove());
+        assertEquals(2, summary.nodeNameToWeightChanges().size());
+
+        var summaryDummyNodeWeights = summary.nodeNameToWeightChanges().get(DUMMY_NODE.getName());
+        assertEquals(10, summaryDummyNodeWeights.weights().shardCount());
+        assertDoublesEqual(20, summaryDummyNodeWeights.weights().diskUsageInBytes());
+        assertDoublesEqual(30, summaryDummyNodeWeights.weights().writeLoad());
+        assertDoublesEqual(40, summaryDummyNodeWeights.weights().nodeWeight());
+        assertEquals(10, summaryDummyNodeWeights.nextWeightsDiff().shardCountDiff());
+        assertDoublesEqual(20, summaryDummyNodeWeights.nextWeightsDiff().diskUsageInBytesDiff());
+        assertDoublesEqual(30, summaryDummyNodeWeights.nextWeightsDiff().writeLoadDiff());
+        assertDoublesEqual(40, summaryDummyNodeWeights.nextWeightsDiff().totalWeightDiff());
+
+        var summarySecondDummyNodeWeights = summary.nodeNameToWeightChanges().get(SECOND_DUMMY_NODE.getName());
+        assertEquals(0, summarySecondDummyNodeWeights.weights().shardCount());
+        assertDoublesEqual(0, summarySecondDummyNodeWeights.weights().diskUsageInBytes());
+        assertDoublesEqual(0, summarySecondDummyNodeWeights.weights().writeLoad());
+        assertDoublesEqual(0, summarySecondDummyNodeWeights.weights().nodeWeight());
+        assertEquals(5, summarySecondDummyNodeWeights.nextWeightsDiff().shardCountDiff());
+        assertDoublesEqual(15, summarySecondDummyNodeWeights.nextWeightsDiff().diskUsageInBytesDiff());
+        assertDoublesEqual(25, summarySecondDummyNodeWeights.nextWeightsDiff().writeLoadDiff());
+        assertDoublesEqual(35, summarySecondDummyNodeWeights.nextWeightsDiff().totalWeightDiff());
+    }
+
+    /**
+     * Helper for double type inputs. assertEquals on double type inputs require a delta.
+     */
+    private void assertDoublesEqual(double expected, double actual) {
+        assertEquals(expected, actual, 00001);
```

Comment:
Am I reading this correctly that the delta is 1? I would have thought we'd be able to make this zero or very close to zero? (perhaps a typo missing a decimal point)